rootex/framework/entity_factory.cpp

```diff
@@ -38,7 +38,7 @@ EntityID EntityFactory::s_CurrentEditorID = -ROOT_ENTITY_ID;
 void EntityFactory::RegisterAPI(sol::state& rootex)
 {
 	sol::usertype<EntityFactory> entityFactory = rootex.new_usertype<EntityFactory>("EntityFactory");
-	entityFactory["Create"] = [](TextResourceFile* t) { return EntityFactory::GetSingleton()->createEntity(t); };
+	entityFactory["Create"] = [](TextResourceFile* t) { return EntityFactory::GetSingleton()->createEntity(JSON::json::parse(t->getString()), t->getPath().generic_string()); };
```

Comment:
I am seeing `JSON::json::parse(something->getString()), something->getPath().generic_string()` everywhere where `createEntity` is called, it'd be better to change the function arguments and do this inside the body of `createEntity`.
Also it would be problematic to create entities like this from Lua